components/core/src/clp_s/search/OutputHandler.hpp

```diff
@@ -70,6 +74,57 @@ class StandardOutputHandler : public OutputHandler {
     void flush() override {}
 };
 
+/**
+ * Output handler that writes to a network destination.
+ */
+class NetworkOutputHandler : public OutputHandler {
+public:
+    // Types
+    class OperationFailed : public TraceableException {
+    public:
+        // Constructors
+        OperationFailed(ErrorCode error_code, char const* const filename, int line_number)
+                : TraceableException(error_code, filename, line_number) {}
+    };
+
+    // Constructors
+    explicit NetworkOutputHandler(
```

Comment:
I feel like we should either connect to the server before constructing the network output handler and just pass in the file descriptor, or provide a method to check if the connection was made successfully. The current design hides connection errors until we write the first result, and potentially hides them completely if no records match the query. Probably easier to debug + less wasteful of compute and IO if we just early exit as soon as we know the connection failed.

Reply:
Which connection errors should we check besides those in the constructor?

Reply:
Only the ones in the constructor I think, unless you can think of something else. I think your current strategy of throwing if we fail during search is fine for everything else I can think of.

Reply:
Do you mean we should check it early? For MongoDB, we also check it in the constructor.

Reply:
Ah nevermind, I see that you're already throwing from the constructor if it fails to connect. LGTM then.